### PR TITLE
LW-7282 idle test hd wallet

### PR DESCRIPTION
**Pull request by mirceahasegan**
# Context

HD wallet load test and test methodology

# Proposed Solution

# Important Changes Introduced

### CODE REVIEW

File: packages/e2e/test/k6/scenarios/idle-clients.js [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+export function scenario_1({ wallets, blockHeightOfLastTx }) {
+  // Get the wallet for the current virtual user
+  // eslint-disable-next-line no-undef
+  const vu = __VU;
+  const wallet = wallets[vu % wallets.length]; // each wallet is a collection of addresses
+  const addresses = wallet.map(({ address }) => address);
+  for (let i = 0; i < NUM_TIP_QUERIES; i++) {
+    cardanoHttpPost(TIP_URL);
+    // No sleep after last query - fetch utxo and tx history immediately
+    if (i < NUM_TIP_QUERIES) {
```
Comment (mkazlauskas) [inline packages/e2e/test/k6/scenarios/idle-clients.js:211]: Isn't this always `true` as it's checked in the `for`?
```suggestion
    if (i + 1 < NUM_TIP_QUERIES) {
```

**Review by mkazlauskas — CHANGES_REQUESTED** [review]

**Review by mkazlauskas — DISMISSED** [review]

File: packages/e2e/test/k6/scenarios/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Running tests locally
```
Comment (iadmytro) [inline packages/e2e/test/k6/scenarios/README.md:1]: better to move it one level higher to `packages/e2e/test/k6`

File: packages/e2e/test/k6/scenarios/idle-clients.js [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+
+/**
+ * Grab the wallets json file to be used by the scenario.
+ * Group the addresses per wallet (single address or HD wallets).
+ */
+export function setup() {
+  console.log(
+    `Ramp-up: ${RAMP_UP_DURATION}; Sustain: ${STEADY_STATE_DURATION}; Poll: ${POLL_INTERVAL}s; Block height change every: ${NUM_TIP_QUERIES}`
+  );
+  const res = http.get(WALLET_ADDRESSES_URL);
```
Comment (iadmytro) [inline packages/e2e/test/k6/scenarios/idle-clients.js:159]: looks like this call will be included in our statistic and there is no way around this so far
https://github.com/grafana/k6/issues/1321

Comment (mirceahasegan) [reply packages/e2e/test/k6/scenarios/idle-clients.js:159]: I will add an inline comment

Comment (mirceahasegan) [reply packages/e2e/test/k6/scenarios/idle-clients.js:160]: done here: [db0bad8](https://github.com/input-output-hk/cardano-js-sdk/pull/808/commits/db0bad8c05b490403264bb49e5a9d101b66067ec)

File: packages/e2e/test/k6/scenarios/README.md (hunk 2 above)
Comment (mirceahasegan) [reply packages/e2e/test/k6/scenarios/README.md]: done here: [6aa3d32](https://github.com/input-output-hk/cardano-js-sdk/pull/808/commits/6aa3d326fccf6275d16b75da47b0d0229a8f0916)

**Review by iadmytro — APPROVED** [review]

**Review by mkazlauskas — APPROVED** [review]
Nice work! :rocket:

**Pull request merged**